crates/common/certificate/src/parse_root_certificate.rs

```diff
@@ -21,14 +21,31 @@ pub fn create_tls_config(
 
 fn new_root_store(cert_path: PathBuf) -> Result<RootCertStore, CertificateError> {
     let mut root_store = RootCertStore::empty();
+    rec_add_root_cert(&mut root_store, cert_path);
+    Ok(root_store)
+}
+
+fn rec_add_root_cert(root_store: &mut RootCertStore, cert_path: PathBuf) {
+    if let Err(err) = try_rec_add_root_cert(root_store, cert_path.clone()) {
+        eprintln!("Ignoring any certificate in {:?} : {}", cert_path, err)
+    }
+}
+
+fn try_rec_add_root_cert(
+    root_store: &mut RootCertStore,
+    cert_path: PathBuf,
+) -> Result<(), CertificateError> {
     if fs::metadata(&cert_path)?.is_dir() {
-        for file_entry in fs::read_dir(cert_path)? {
-            add_root_cert(&mut root_store, file_entry?.path())?;
+        for file_entry in fs::read_dir(cert_path)?.flatten() {
+            rec_add_root_cert(root_store, file_entry.path());
         }
-    } else {
-        add_root_cert(&mut root_store, cert_path)?;
+    } else if let Err(err) = add_root_cert(root_store, cert_path.clone()) {
+        eprintln!(
+            "Ignoring invalid certificate chain {:?} : {}",
```

Comment:
```suggestion
            "Failed to add {:?} certificate to rootstore because of an eror : {}",
```

Reply:
The `error` will contain why it failed to add to the `root store` right?, then we can just print that error and the path to the certificate, instead of we specifically saying it has `invalid certificate chain`

Reply:
I agree that the certificate chain might be simply not accessible. However, I prefer to use the word `ignoring` rather than `failed` because the function proceed with the other certificate files.

I propose:

```
            "Ignoring certificates in file {:?} due to {}",
```

Reply:
ok